### ISSUE TITLE
TypeModel generates wrong XMLArray Namespace

### ISSUE THREAD

**Issue by LokiMidgard** [mganss/XmlSchemaClassGenerator#28]
I use Elements from other namespaces and include those in my type using the ```ref```-Attribute.
The Daten Element is in the namespace ```http://nota-game.azurewebsites.net/schema/nota``` 

```
  <element name="Daten">
    <complexType>
      <sequence>
        <element minOccurs="0" name="Pfade">
          <complexType>
             ....
          </complexType>
        </element>
        <element ref="lebewesen:Lebewesene"/>
        ...
      </sequence>
    </complexType>
  </element>

```
The generated Property looks like:

```
        /// <summary>
        /// </summary>
        [System.Xml.Serialization.XmlArrayAttribute("Lebewesene", Namespace="http://nota-game.azurewebsites.net/schema/nota")]
        [System.Xml.Serialization.XmlArrayItemAttribute("Lebewesen", Namespace="http://nota-game.azurewebsites.net/schema/lebewesen")]
        public System.Collections.ObjectModel.Collection<Nota.CharacterGenerator.CoreData.Lebewesen.LebeweseneLebewesen> Lebewesene { get {...} }

```

However the Namespace of the XmlArrayAttribute should be ```http://nota-game.azurewebsites.net/schema/lebewesen```.

The TypeModel sets the namespace to the Owning type if Form equals ```XmlSchemaForm.Qualified```. However if the element is not delcread at this position but referenced from another Namespace it should be the namespace of the referenced element. In your code this seems to be XmlNamespace.

I tested to change the order of the checks in TypeModel.
First Checking if ```XmlNamespace != null``` and if this fails checking if ```Form == XmlSchemaForm.Qualified```.

This worked for my so far, but I'm not sure about the side effects. That's why I haven't created an pull request yet.

[This are the changes.](https://github.com/LokiMidgard/XmlSchemaClassGenerator/commit/5338b4f02f9a85931e7e435438f4571c6d1d3ca2)

**Comment by mganss** [comment]
If the unit tests pass please go ahead and make a PR.

**Comment by mganss** [comment]
Fixed by #29
